gunicorn/sock.py

```diff
@@ -38,6 +38,11 @@ def __getattr__(self, name):
 
     def set_options(self, sock, bound=False):
         sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        try:
+            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        except (AttributeError, socket.error) as e:
```

Comment:
I'd rather see a check like `if hasattr(socket, 'SO_REUSEPORT'):`. If kernel doesn't support `SO_REUSEPORT` option, there is no point to try and print a warning.